### DISCUSSION TITLE
Before install

### DISCUSSION THREAD

**Discussion by xxlimit** [amidaware/tacticalrmm#1205] [Q&A]
- OS: [Debian 11]

**Installation Method:**
  - [x] Standard

How do you setup the listening port and certs before install?

**Comment by dinger1986** [comment]
Check out docs.tacticalrmm.com

The install script setups up certs and the docs take you through opening ports

**Comment by xxlimit** [comment]
Can you give a more in depth example of where I can choose the certs and ports in the install file?

**Reply by dinger1986** [reply]
I have no idea what you mean. 

What are you trying to achieve?


The docs takes you through the setup

**Reply by wh1te909** [reply]
I think he wants this https://docs.tacticalrmm.com/functions/settings_override/#using-your-own-wildcard-ssl-cert

There are many internal listening ports, you don't really need to configure these. The only public port that should be exposed should be 443 and you can choose to do that however you want, either via UFW firewall on the vm, or if you have your own firewall in front of the VM you can do it there

**Reply by xxlimit** [reply]
That's after install, I just wanna be able to configure the listening port and certs before install.

**Reply by dinger1986** [reply]
Do you mean change the install script? As you can't configure something that doesn't exist lol. Do you have a cert?

**Reply by dinger1986** [reply]
Again what are you trying to achieve what's your end goal here?

**Reply by xxlimit** [reply]
To host on tor to view my home computers without port forwarding.

**Reply by wh1te909** [reply]
You would need to read through the install script and edit it. For example these 2 lines https://github.com/amidaware/tacticalrmm/blob/d5a6063e5ed8bf951a415af9e3cf039778c899d3/install.sh#L167 edit with path to your certs then make sure to also added them in `local_settings.py` using the link above. But yes as @dinger1986 you can't change stuff before it exists so you would need to make those changes yourself in the install script.

**Reply by xxlimit** [reply]
How would you change the listening port before install?

**Reply by dinger1986** [reply]
This has already been answered.

1. You can't change something that doesnt exist.
2. Read through the install script and look at the nginx configs.

**Reply by xxlimit** [reply]
So the listening port is unconfigurable, yet you can edit the cert.

**Reply by dinger1986** [reply]
I have no idea what you are on about. You can change the ports in the nginx configs if you wanted, it might not work but you can try it

**Reply by xxlimit** [reply]
If you look in the install.sh 443 is very hard to change.

**Reply by xxlimit** [reply]
Any ideas of how to fix this?
Because it seems letsencrypt hates tor.
![16574051710324685848628047842050](https://user-images.githubusercontent.com/66466672/178124399-ea888dc3-4d57-4189-ba47-e3a57a6448f0.jpg)

**Reply by dinger1986** [reply]
I would assume letsencrypt doesn't support your url/domain.

As the docs say it needs to be a real domain.

You could maybe make up your own ssl certs and get nats working with that but you are miles away from a standard install it's really on you to figure it out if you want to achieve that.

**Reply by xxlimit** [reply]
I already changed cert in the install script.
![16574058572563446912116885525471](https://user-images.githubusercontent.com/66466672/178124665-e232a057-4a67-4ece-a530-a65560065d08.jpg)

**Reply by dinger1986** [reply]
So you would need to change the script so it knows what's it's looking for and disable letsencrypt.

If you are unsure maybe install normally then change it later.

Doing something noones done before so write it once you have it working. It's needing fairly technical changes for some of it.

Good luck with this,it's really miles out of scope to be supporting this.

**Reply by xxlimit** [reply]
I already fixed that now it's making a user on Django.
![1657408742995802212655369731616](https://user-images.githubusercontent.com/66466672/178125583-20520a8d-07cc-4314-a6ff-de98c85e5975.jpg)
